### PR TITLE
lastFriarsXNC Properties Support

### PR DESCRIPTION
**Pull request by JamesDowney**
@horrible-little-slime blessed us with a means to track the `turns_spent` in a friars zone when one of the quest NCs is hit. This utilizes those properties to generate more accurate estimations of the turns remaining in friar zones.

I also fixed the capitalization of the friar locations in the `Level 6.ash` file to match Mafia.

### CODE REVIEW

File: Source/relay/TourGuide/Quests/Level 6.ash [hunk 1]
```diff
@@ -72,12 +73,12 @@ float QLevel6TurnsToCompleteArea(location place)
         turns_remaining = 10000.0; //how do you refer to infinity in this language?
 
     int max_turns_remaining = ncs_remaining * 5;
-    if (place == $location[the dark neck of the woods])
```
Comment (docrostov) [inline Source/relay/TourGuide/Quests/Level 6.ash:75]: we should probably actually delete the code block from base_quest_state re: "turns on last nc" if we're just gonna use the lastFriarsXNC prefs going forward. also might want to push required version # but that's a whole other can of worms.

File: Source/relay/TourGuide/Quests/Level 6.ash [hunk 2]
```diff
@@ -23,13 +23,20 @@ void QLevel6Init()
 float QLevel6TurnsToCompleteArea(location place)
 {
     //FIXME not sure how accurate these calculations are.
-    int turns_spent_in_zone = turnsAttemptedInLocation(place); //not always accurate
     int ncs_found = noncombatTurnsAttemptedInLocation(place);
 
+
+    // get_property("lastFriarsXNC").to_int() will return 0 until the first NC is hit, then it will return the turns spent in the zone prior to hitting the NC,
```
Comment (docrostov) [inline Source/relay/TourGuide/Quests/Level 6.ash:29]: i think i get this but can we add an example? just to add some clarity? suggested ex, and please fix this if i misread the comment, as i think this describes what you described:

```
// For example, if you spend 3 turns in the neck, hit an NC, then hit another NC after 2 turns the pref will manifest as:
//     TURN 1: lastFriarsNeckNC = 0  
//     TURN 2: lastFriarsNeckNC = 0
//     TURN 3: lastFriarsNeckNC = 0
//     TURN 4: lastFriarsNeckNC = 0 <= hits NC
//     TURN 5: lastFriarsNeckNC = 1
//     TURN 6: lastFriarsNeckNC = 2
//     TURN 7: lastFriarsNeckNC = 3 <= hits NC
```

**Review by docrostov — APPROVED** [review]
sorry this took so long to approve. overall it seems fine. i wish i understood how the pref worked a bit better; the comment in here on line 28-30 is helpful but i feel like a practical example would be nice. i will merge it up tomorrow probably. thanks for putting in the time bud.

**Comment by docrostov** [comment]
actually before we merge this i have one thought -- can we add a small note if they are at 5 turns since the last NC and thus due an NC on their next turn? that's a nice optimization that's more useful than normal since it means they have a queued up push turn for breathitin

File: Source/relay/TourGuide/Quests/Level 6.ash (hunk 1 above)
Comment (JamesDowney) [reply Source/relay/TourGuide/Quests/Level 6.ash:75]: You're definitely right on both accounts. I think I put this together before we had the conversation about maintaining backwards compatibility and bumping the required version, so it wasn't on my mind but it should be done

File: Source/relay/TourGuide/Quests/Level 6.ash (hunk 2 above)
Comment (JamesDowney) [reply Source/relay/TourGuide/Quests/Level 6.ash:29]: You got it, for sure. I remember this being a little weird to puzzle out so making it as clear as possible to anyone else that looks at the code is definitely a good idea.

**Comment by JamesDowney** [comment]
> actually before we merge this i have one thought -- can we add a small note if they are at 5 turns since the last NC and thus due an NC on their next turn? that's a nice optimization that's more useful than normal since it means they have a queued up push turn for breathitin

I'm genuinely just too unfamiliar with CMC to feel like I've got a good handle on how best to tackle this. Would `Your next adventure here will be an NC` in bold at the bottom of the location's section be sufficient?

**Pull request merged**